prime-router/docs/design/validation/design.md

```diff
@@ -0,0 +1,218 @@
+# Problem Statement
+As a sender I want submitted data to be validated to ensure data integrity and quality. As a receiver I want received
+data to be validated to ensure data integrity and quality. For this to happen, there needs to exist a validation 
+setting as well as reusable validation components to perform the sending and receiving validation. 
+
+### Shared Components
+To facilitate reuse and extensibility, a new setting type `validationProfile` shall be created representing a type of
+validation, the validations to perform, and any applicable configuration parameters. This setting is reusable because
+senders and receivers can only be set up to receive one format at a time, so there should not be confusion over which
+validationProfile pertains to which format.
+
+When a message fails validation. Validation errors should show up in:
+- The submission history API
+- The Action Log in the database in such a way that the engagement team can easily query for it
+- Notify the sender, if appropriate
+
+Tickets needed for both epics:
+- Create the validation profile setting: https://github.com/CDCgov/prime-reportstream/issues/9161 <br/>
+- Add validation errors to the submission history API: https://github.com/CDCgov/prime-reportstream/issues/9036 <br/>
+- Add validation errors to the action log: https://github.com/CDCgov/prime-reportstream/issues/9221 <br/>
+
+## FHIR Validation
+### Proposed Design
+The diagram below proposes four validation "checkpoints":
+1. Upon receipt of a submission in the SenderReceiver
+2. Per bundle after debatch in the convert function
```

Comment:
I'm wondering if we need this step? Wouldn't it be easier just to do one validation step after enrichment to see if we have everything that we need?

Reply:
@thetaurean Please address